File: Assets/Typescript~/src/Client/Signals/EntityDeathClientSignal.ts
import { DamageType } from "@Easy/Core/Shared/Damage/DamageType";
import { Entity } from "@Easy/Core/Shared/Entity/Entity";

export class EntityDeathClientSignal {
	constructor(
		public readonly entity: Entity,
		public readonly damageType: DamageType,
		public readonly fromEntity?: Entity,
	) {}
}

File: Assets/Bundles/Client/Resources/TS/Signals/EntityDeathClientSignal.lua
-- Compiled with unity-ts v2.1.0-75
local EntityDeathClientSignal
do
	EntityDeathClientSignal = setmetatable({}, {
		__tostring = function()
			return "EntityDeathClientSignal"
		end,
	})
	EntityDeathClientSignal.__index = EntityDeathClientSignal
	function EntityDeathClientSignal.new(...)
		local self = setmetatable({}, EntityDeathClientSignal)
		return self:constructor(...) or self
	end
	function EntityDeathClientSignal:constructor(entity, damageType, fromEntity)
		self.entity = entity
		self.damageType = damageType
		self.fromEntity = fromEntity
	end
end
return {
	EntityDeathClientSignal = EntityDeathClientSignal,
}
-- ----------------------------------
-- ----------------------------------
